### PR TITLE
feat: Making this responsive 

### LINKED ISSUE #1: Mobile View
The societytable in its current form does not render on mobile screens. It would be good to make it responsive.

### PR DESCRIPTION
**Pull request by SachdevJai**
# Description

This pr is for making this application responsive as it currently breaks for smaller screens

Fixes #1 

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

Using the developer tools in chrome firefox and edge to test responsiveness

**Test Configuration**:
* Firmware version:
* Hardware:
* Toolchain:
* SDK:

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: package-lock.json
Comment (harshkhandeparkar) [inline package-lock.json:1]: Please use `pnpm` not `npm`.

File: package.json [hunk 1]
```diff
@@ -10,8 +10,10 @@
     "preview": "vite preview"
   },
   "dependencies": {
+    "pnpm": "^9.1.4",
```
Comment (harshkhandeparkar) [inline package.json]: Shouldn't be added here.

File: src/components/App.tsx [hunk 2]
```diff
@@ -12,7 +11,6 @@ function App() {
           <Header />
         </div>
         <div className="row table-container">
-          <Legend />
```
Comment (harshkhandeparkar) [inline src/components/App.tsx:15]: If you are moving the legend, you can increase the width of the table.

File: src/components/Table.tsx [hunk 3]
```diff
@@ -42,7 +43,6 @@ function TableCell(props: TableCellProps) {
 				</a>
 				<Tooltip className="soc-tooltip" id={`${props.index.toString()}-${props.society.symbol}`} delayHide={0} delayShow={0}>
 					<span className="soc-tooltip-name">{props.society.name}</span>
-					<p className="soc-toolip-desc">{props.society.description}</p>
```
Comment (harshkhandeparkar) [inline src/components/Table.tsx:45]: ?

Comment (SachdevJai) [reply src/components/Table.tsx:45]: in the thread we discussed that tooltip will only show the name and there will be a modal for the desc?

Comment (harshkhandeparkar) [reply src/components/Table.tsx:45]: Is a tooltip useful if we have a modal?

File: src/components/Table.tsx [hunk 4]
```diff
@@ -1,90 +1,17 @@
-import { Tooltip } from 'react-tooltip';
+// import { Block, ISociety } from "../data/societies";
```
Comment (harshkhandeparkar) [inline src/components/Table.tsx]: Remove instead of commenting.

File: src/components/Tablecell.tsx [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+	const NUMERICAL_ROOTS = ["nil", "un", "bi", "tri", "quad", "pent", "hex", "sept", "oct", "en"];
+	const index_digits = index.toString().split('').map((digit_str) => parseInt(digit_str));
+
+	let symbol = index_digits.map((digit) => NUMERICAL_ROOTS[digit][0]).join('');
+	symbol = symbol[0].toUpperCase() + symbol.slice(1);
+
+	let name = index_digits.map((digit) => NUMERICAL_ROOTS[digit]).join('');
+	name.replace('nnn', 'nn');
+	name.replace('iii', 'ii');
+	name = name[0].toUpperCase() + name.slice(1) + "ium";
```
Comment (harshkhandeparkar) [inline src/components/Tablecell.tsx:25]: Can you move this statement up before the two `.replace()` statements? This was a bug I just realized :stuck_out_tongue:

File: src/styles/_header.scss [hunk 6]
```diff
@@ -11,25 +11,49 @@
...
 
+
 .ribbon {
 	font-size: 20px !important;
 	/* This ribbon is based on a 16px font side and a 24px vertical rhythm. I've used em's to position each element for scalability. If you want to use a different font size you may have to play with the position of the ribbon elements */
 	width: 40%;
 	position: relative;
 	font-family: 'Nixie One';
-	font-size: 6em;
+	// font-size: 6em;
```
Comment (harshkhandeparkar) [inline src/styles/_header.scss]: please just remove instaed of commenting.

File: src/styles/_header.scss [hunk 7]
```diff
@@ -11,25 +11,49 @@
...
 	/* Based on 24px vertical rhythm. 48px bottom margin - normally 24 but the ribbon 'graphics' take up 24px themselves so we double it. */
 
+	@media screen and (max-width: 768px){
+		font-size: 1.1rem !important;
+		width: 25%;
+		padding: 1em 5em;
+		margin-bottom: 2em;
+	}
+
+	// @media screen and (max-width: 1024px){
```
Comment (harshkhandeparkar) [inline src/styles/_header.scss]: yeet.

File: src/components/Tablecell.tsx [hunk 8]
```diff
@@ -0,0 +1,140 @@
...
+				<Tooltip className="soc-tooltip" id={`${props.index.toString()}-${symbol}`} delayHide={0} delayShow={0}>
+					<span className="soc-tooltip-name">{name} (Not Discovered)</span>
+					<p className="soc-toolip-desc">This society has not been discovered yet and has been assigned a temporary name.</p>
+				</Tooltip>
+			</>;
+	}
+}
+
+function TableCell(props: TableCellProps) {
+    Modal.setAppElement('#root');
```
Comment (harshkhandeparkar) [inline src/components/Tablecell.tsx:84]: If this is a global setting, set it in the `main.tsx`.

File: src/components/Tablecell.tsx [hunk 9]
```diff
@@ -0,0 +1,140 @@
...
+        if(!modalIsOpen) setIsOpen(true);
+    }
+    
+    
+    function closeModal() {
+        if(modalIsOpen) setIsOpen(false);
+    }
+
+    
+    return <div onClick={openModal}>
```
Comment (harshkhandeparkar) [inline src/components/Tablecell.tsx]: Make this a `button` element or add a hover effect to change the cursor type to the clickable version.

File: src/components/Tablecell.tsx [hunk 10]
```diff
@@ -0,0 +1,140 @@
...
+        <Cell {...props}/>
+        {props.type === 'society' && <Modal
+            isOpen={modalIsOpen}
+            onRequestClose={closeModal}
+            shouldCloseOnEsc={true}
+            contentLabel="Society Details"
+            style={{
+                overlay: {
```
Comment (harshkhandeparkar) [inline src/components/Tablecell.tsx]: Move to scss if possible.

File: src/components/Tablecell.tsx [hunk 11]
```diff
@@ -0,0 +1,140 @@
...
+                    <Cell {...props} />
+                    <p>{props.society.description}</p>
+                    <a href={props.society.link} target="_blank" rel="noreferrer"><img style={{ height: '1.5em', width: '1.5em' }} src={facebook} alt="Facebook" /></a>
+                </div>
+            </Modal>
+        }
+    </div>
+
+}
+
```
Comment (harshkhandeparkar) [inline src/components/Tablecell.tsx:137]: Run all files through a formatter once.

File: src/components/Tablecell.tsx [hunk 12]
```diff
@@ -0,0 +1,140 @@
...
+                    width: '80%',
+                    maxWidth: '500px',
+                }
+            }}
+            >
+                <div className='table soc-modal'>
+                    <h2>{props.society.name}</h2>
+                    <Cell {...props} />
+                    <p>{props.society.description}</p>
+                    <a href={props.society.link} target="_blank" rel="noreferrer"><img style={{ height: '1.5em', width: '1.5em' }} src={facebook} alt="Facebook" /></a>
```
Comment (harshkhandeparkar) [inline src/components/Tablecell.tsx]: Again, move styles to the scss files.

File: src/components/Tablecell.tsx [hunk 13]
```diff
@@ -0,0 +1,140 @@
...
+	name.replace('iii', 'ii');
+	name = name[0].toUpperCase() + name.slice(1) + "ium";
+
+	return {
+		name,
+		symbol
+	}
+}
+
+function Cell(props: TableCellProps & { insideModal?: boolean }) {
```
Comment (harshkhandeparkar) [inline src/components/Tablecell.tsx]: Can rename this component to `Element`

File: src/components/Tablecell.tsx [hunk 14]
```diff
@@ -0,0 +1,140 @@
...
+function getIUPACTemporaryNameAndSymbol(index: number): { name: string, symbol: string } {
+	const NUMERICAL_ROOTS = ["nil", "un", "bi", "tri", "quad", "pent", "hex", "sept", "oct", "en"];
+	const index_digits = index.toString().split('').map((digit_str) => parseInt(digit_str));
+
+	let symbol = index_digits.map((digit) => NUMERICAL_ROOTS[digit][0]).join('');
+	symbol = symbol[0].toUpperCase() + symbol.slice(1);
+
+	let name = index_digits.map((digit) => NUMERICAL_ROOTS[digit]).join('');
+	name.replace('nnn', 'nn');
+	name.replace('iii', 'ii');
```
Comment (harshkhandeparkar) [inline src/components/Tablecell.tsx]: This should be `name.replace('ii', 'i');`

File: src/components/Table.tsx [hunk 15]
```diff
@@ -1,90 +1,17 @@
...
+import TableCell from './Tablecell';
 import { Block, ISociety } from "../data/societies";
-import { Grid, GridItem } from "./Grid";
+
 
 /**
  * Gives the temporary name and symbol of an undiscovered element according to IUPAC nomenclature.
  * See https://en.wikipedia.org/wiki/Systematic_element_name for more information
  * @param index The index of the undiscovered element
  * @returns The temporary name and symbol
```
Comment (harshkhandeparkar) [inline src/components/Table.tsx]: Bhaiyya move this comment as well

**Review by harshkhandeparkar — CHANGES_REQUESTED** [review]

**Review by harshkhandeparkar — APPROVED** [review]

**Pull request merged**